unit_tests/RecordUnitTests.d

```diff
@@ -524,3 +524,25 @@ D");
 
 }
 
+@("issue #foo - translate record with name from macro")
+unittest {
+    assertTranslates(
+    q{
+
+#define __FSID_T_TYPE struct { int __val[2]; }
+typedef  __FSID_T_TYPE __fsid_t;
+typedef __fsid_t fsid_t;
+
+},
+    q{extern (C):
```

Comment:
`q{}` gets highlighted as code, which is what these strings are. `q"C` and `q"D` don't. One might object to using `q{}` for C code but the syntax is so similar that in practice it works (at least it does in Emacs).